packages/responsive/src/useMediaQueries.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { FC, useEffect, useState } from "react"
+import { useTheme } from "@kaizen/design-tokens"
+
+type Props = { [key: string]: string }
+
+export const subtractOnePixel = (breakpoint: string) =>
+  `${parseInt(breakpoint, 10) - 1}px`
+
+export const useMediaQueries = (
+  propQueries: Props = {}
+): {
+  queries: {
+    [key: string]: boolean
+    isSmall: boolean
+    isMedium: boolean
+    isLarge: boolean
+    isMediumOrSmaller: boolean
+    isMediumOrLarger: boolean
+  }
+  components: {
+    [key: string]: FC
+    SmallOnly: FC
+    MediumOnly: FC
+    LargeOnly: FC
+    MediumOrSmaller: FC
+    MediumOrLarger: FC
+  }
+} => {
+  const theme = useTheme()
+
+  // ---------------------------------------
+  // Create Kaizen breakpoint matches for initial state
+  // ---------------------------------------
+  const smallMatchMedia = window.matchMedia(
+    `(max-width: ${subtractOnePixel(theme.layout.breakpoints.medium)})`
+  )
+  const mediumMatchMedia = window.matchMedia(
+    `(min-width: ${
+      theme.layout.breakpoints.medium
+    }) and (max-width: ${subtractOnePixel(theme.layout.breakpoints.large)})`
+  )
+  const largeMatchMedia = window.matchMedia(
+    `(min-width: ${theme.layout.breakpoints.large})`
+  )
```

Comment:
Probably nothing to do, but it's hard to say how often these are going to get called (probably not too often but perhaps if someone used this hook in a table component with lots of cells/rows etc).
It's worth noting the performance difference when memoizing this: https://jsbench.me/sbkqzzsdpn/1
Probably not worth prematurely optimising, but if you wanted to now or in the future, you could wrap these with `React.useMemo`

Reply:
We’ll potentially end up with a ton of these in the future, so some optimisation is probably worthwhile on a proactive level. We’d need to trigger the `useMemo` based on a resize handler, yeah?

Reply:
i think it's much simpler: 
```
const largeMatchMedia = React.useMemo(() => window.matchMedia(
    `(min-width: ${theme.layout.breakpoints.large})`
  ), [theme.layout.breakpoints.large])
```

In other words: `largeMatchMedia` is a computed value that depends on `theme.layout.breakpoints.large` - only re-calculate it when `theme.layout.breakpoints.large` changes (which is, like, never)

Reply:
i think we should be cautious of bikeshedding: this hook could be optimised in a number of ways.
we don't really have any evidence to support decisions, only good guesses really.

now i'm thinking i might setup a story with heaps of instances of it to test it...

Reply:
> i think it's much simpler

Ah, you’re right. I wasn’t really sure about what `matchMedia` returns but it’s just an instance so we should really be fine.